crates/orchestrator/src/network_spec/node.rs

```diff
@@ -0,0 +1,223 @@
+use configuration::shared::{
+    node::{EnvVar, NodeConfig},
+    resources::Resources,
+    types::{Arg, AssetLocation, Command, Image},
+};
+use multiaddr::Multiaddr;
+
+use crate::{
+    errors::OrchestratorError,
+    generators,
+    network::AddNodeOpts,
+    shared::types::{ChainDefaultContext, NodeAccounts, ParkedPort},
+};
+
+/// A node configuration, with fine-grained configuration options.
+#[derive(Debug, Clone)]
+pub struct NodeSpec {
+    /// Node name (should be unique or an index will be appended).
```

Comment:
nit: Fields are private, doc can be transformed into simple comment

```suggestion
    // Node name (should be unique or an index will be appended).
```

Reply:
Yes, nice catch 👍